commit 7948348cdc115389700242901c91d323192850a8
Author: Willy Tarreau <w@1wt.eu>
Date:   Fri Mar 29 17:26:33 2019 +0100

    BUILD: use inttypes.h instead of stdint.h
    
    I found on an (old) AIX 5.1 machine that stdint.h didn't exist while
    inttypes.h which is expected to include it does exist and provides the
    desired functionalities.
    
    As explained here, stdint being just a subset of inttypes for use in
    freestanding environments, it's probably always OK to switch to inttypes
    instead:
    
      https://pubs.opengroup.org/onlinepubs/009696799/basedefs/stdint.h.html
    
    Also it's even clearer here in the autoconf doc :
    
      https://www.gnu.org/software/autoconf/manual/autoconf-2.61/html_node/Header-Portability.html
    
      "The C99 standard says that inttypes.h includes stdint.h, so there's
       no need to include stdint.h separately in a standard environment.
       Some implementations have inttypes.h but not stdint.h (e.g., Solaris
       7), but we don't know of any implementation that has stdint.h but not
       inttypes.h"
    
    (cherry picked from commit a1bd1faeebd03825677d111a1350ee04d625f165)
    [wt: dropped include/proto/protocol_buffers.h]
    Signed-off-by: Willy Tarreau <w@1wt.eu>
    (cherry picked from commit 19f364bb4ad0205e134dc527e5164b7a21c2ad07)
    [cf: missing files removed]
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/contrib/hpack/gen-rht.c b/contrib/hpack/gen-rht.c
index b1b90317..4260ffbe 100644
--- a/contrib/hpack/gen-rht.c
+++ b/contrib/hpack/gen-rht.c
@@ -9,7 +9,7 @@
  *   00 => 0x0a, 01 => 0x0d, 10 => 0x16, 11 => EOS
  */
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/contrib/plug_qdisc/plug_qdisc.c b/contrib/plug_qdisc/plug_qdisc.c
index 294994eb..606a834c 100644
--- a/contrib/plug_qdisc/plug_qdisc.c
+++ b/contrib/plug_qdisc/plug_qdisc.c
@@ -1,4 +1,4 @@
-#include <stdint.h>
+#include <inttypes.h>
 #include <netlink/cache.h>
 #include <netlink/cli/utils.h>
 #include <netlink/cli/tc.h>
diff --git a/contrib/spoa_example/include/spop_functions.h b/contrib/spoa_example/include/spop_functions.h
index 8319e41b..cff45321 100644
--- a/contrib/spoa_example/include/spop_functions.h
+++ b/contrib/spoa_example/include/spop_functions.h
@@ -1,7 +1,7 @@
 #ifndef _SPOP_FUNCTIONS_H
 #define _SPOP_FUNCTIONS_H
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <string.h>
 #include <spoe_types.h>
 
diff --git a/contrib/wireshark-dissectors/peers/packet-happp.c b/contrib/wireshark-dissectors/peers/packet-happp.c
index a1983316..6fca353a 100644
--- a/contrib/wireshark-dissectors/peers/packet-happp.c
+++ b/contrib/wireshark-dissectors/peers/packet-happp.c
@@ -22,7 +22,7 @@
  */
 
 #include <stdio.h>
-#include <stdint.h>
+#include <inttypes.h>
 #include <inttypes.h>
 #include <arpa/inet.h>
 
diff --git a/include/common/hpack-dec.h b/include/common/hpack-dec.h
index b03398a4..bea2d521 100644
--- a/include/common/hpack-dec.h
+++ b/include/common/hpack-dec.h
@@ -28,7 +28,7 @@
 #ifndef _COMMON_HPACK_DEC_H
 #define _COMMON_HPACK_DEC_H
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <common/chunk.h>
 #include <common/config.h>
 #include <common/hpack-tbl.h>
diff --git a/include/common/hpack-enc.h b/include/common/hpack-enc.h
index 0a44dfc7..fee56fd4 100644
--- a/include/common/hpack-enc.h
+++ b/include/common/hpack-enc.h
@@ -28,7 +28,7 @@
 #ifndef _COMMON_HPACK_ENC_H
 #define _COMMON_HPACK_ENC_H
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <common/chunk.h>
 #include <common/config.h>
 #include <common/ist.h>
diff --git a/include/common/hpack-huff.h b/include/common/hpack-huff.h
index 85ca4171..04276d2c 100644
--- a/include/common/hpack-huff.h
+++ b/include/common/hpack-huff.h
@@ -27,7 +27,7 @@
 #ifndef _PROTO_HPACK_HUFF_H
 #define _PROTO_HPACK_HUFF_H
 
-#include <stdint.h>
+#include <inttypes.h>
 
 int huff_enc(const char *s, char *out);
 int huff_dec(const uint8_t *huff, int hlen, char *out, int olen);
diff --git a/include/common/hpack-tbl.h b/include/common/hpack-tbl.h
index 2cbc2bf6..ca3f2aa9 100644
--- a/include/common/hpack-tbl.h
+++ b/include/common/hpack-tbl.h
@@ -27,7 +27,7 @@
 #ifndef _COMMON_HPACK_TBL_H
 #define _COMMON_HPACK_TBL_H
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <stdlib.h>
 #include <common/config.h>
 #include <common/http-hdr.h>
diff --git a/include/common/http-hdr.h b/include/common/http-hdr.h
index a0bb341b..db1bfa14 100644
--- a/include/common/http-hdr.h
+++ b/include/common/http-hdr.h
@@ -27,7 +27,7 @@
 #ifndef _COMMON_HTTP_HDR_H
 #define _COMMON_HTTP_HDR_H
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <common/ist.h>
 
 /* a header field made of a name and a value. Such structure stores 4 longs so
diff --git a/include/proto/shctx.h b/include/proto/shctx.h
index 55cb2a77..3b42ed3b 100644
--- a/include/proto/shctx.h
+++ b/include/proto/shctx.h
@@ -17,7 +17,7 @@
 #include <common/mini-clist.h>
 #include <types/shctx.h>
 
-#include <stdint.h>
+#include <inttypes.h>
 
 #ifndef USE_PRIVATE_CACHE
 #ifdef USE_PTHREAD_PSHARED
diff --git a/src/h2.c b/src/h2.c
index 5c83d6b6..d7a03405 100644
--- a/src/h2.c
+++ b/src/h2.c
@@ -25,7 +25,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <common/config.h>
 #include <common/h2.h>
 #include <common/http-hdr.h>
diff --git a/src/hpack-dec.c b/src/hpack-dec.c
index 3fd4224d..ad5b23a8 100644
--- a/src/hpack-dec.c
+++ b/src/hpack-dec.c
@@ -25,7 +25,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/src/hpack-enc.c b/src/hpack-enc.c
index cb0767ed..685c9f3d 100644
--- a/src/hpack-enc.c
+++ b/src/hpack-enc.c
@@ -25,7 +25,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/src/hpack-huff.c b/src/hpack-huff.c
index cbf1fa02..bdcff7fe 100644
--- a/src/hpack-huff.c
+++ b/src/hpack-huff.c
@@ -26,7 +26,7 @@
  */
 
 #include <stdio.h>
-#include <stdint.h>
+#include <inttypes.h>
 #include <string.h>
 
 #include <common/config.h>
diff --git a/src/hpack-tbl.c b/src/hpack-tbl.c
index e2d4426f..21aa4bc8 100644
--- a/src/hpack-tbl.c
+++ b/src/hpack-tbl.c
@@ -25,7 +25,7 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
-#include <stdint.h>
+#include <inttypes.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/src/sha1.c b/src/sha1.c
index 3b562b55..b7c2d709 100644
--- a/src/sha1.c
+++ b/src/sha1.c
@@ -26,7 +26,7 @@
 
 /* this is only to get definitions for memcpy(), ntohl() and htonl() */
 #include <string.h>
-#include <stdint.h>
+#include <inttypes.h>
 #include <arpa/inet.h>
 
 #include <import/sha1.h>
diff --git a/src/xxhash.c b/src/xxhash.c
index 5702e64f..4792f128 100644
--- a/src/xxhash.c
+++ b/src/xxhash.c
@@ -98,7 +98,7 @@ static void* XXH_memcpy(void* dest, const void* src, size_t size)
 // Basic Types
 //**************************************
 #if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L   // C99
-# include <stdint.h>
+# include <inttypes.h>
 typedef uint8_t  BYTE;
 typedef uint16_t U16;
 typedef uint32_t U32;
